dom-creation/main.ts

```diff
@@ -0,0 +1,108 @@
+    imageUrl: 'images/bulbasaur.png',
+  },
+  {
+    number: '004',
+    name: 'Charmander',
+    description:
+      'It has a preference for hot things. When it rains, steam is said to spout from the tip of its tail.',
+    imageUrl: 'images/charmander.png',
+  },
+  {
+    number: '007',
+    name: 'Squirtle',
+    description:
+      'When it retracts its long neck into its shell, it squirts out water with vigorous force.',
+    imageUrl: 'images/squirtle.png',
+  },
+  {
+    number: '002',
+    name: 'Ivysaur',
+    description:
+      'When the bulb on its back grows large, it appears to lose the ability to stand on its hind legs.',
+    imageUrl: 'images/ivysaur.png',
+  },
+  {
+    number: '005',
+    name: 'Charmeleon',
+    description:
+      'It has a barbaric nature. In battle, it whips its fiery tail around and slashes away with sharp claws.',
+    imageUrl: 'images/charmeleon.png',
+  },
+  {
+    number: '008',
+    name: 'Wartortle',
+    description:
+      'It is recognized as a symbol of longevity. If its shell has algae on it, that Wartortle is very old.',
+    imageUrl: 'images/wartortle.png',
+  },
+  {
+    number: '003',
+    name: 'Venusaur',
+    description:
+      'Its plant blooms when it is absorbing solar energy. It stays on the move to seek sunlight.',
+    imageUrl: 'images/venusaur.png',
+  },
+  {
+    number: '006',
+    name: 'Charizard',
+    description:
+      'It spits fire that is hot enough to melt boulders. It may cause forest fires by blowing flames.',
+    imageUrl: 'images/charizard.png',
+  },
+  {
+    number: '009',
+    name: 'Blastoise',
+    description:
+      'It crushes its foe under its heavy body to cause fainting. In a pinch, it will withdraw inside its shell.',
+    imageUrl: 'images/blastoise.png',
+  },
+];
+
+function renderPokemon(pokemon: Pokemon): HTMLDivElement {
+  const $outerDiv = document.createElement('div');
+  $outerDiv.classList.add('column-third');
+  const $nestedDiv1 = document.createElement('div');
+  $nestedDiv1.classList.add('pokemon-card');
+  const $img = document.createElement('img');
+  $img.setAttribute('src', pokemon.imageUrl);
+  const $nestedDiv2 = document.createElement('div');
+  $nestedDiv2.classList.add('pokemon-card-text');
+  const $h2 = document.createElement('h2');
+  $h2.textContent = pokemon.name;
+  const $h3 = document.createElement('h3');
+  $h3.textContent = pokemon.number;
+  const $p = document.createElement('p');
+  $p.textContent = pokemon.description;
+
+  $nestedDiv2.appendChild($h2);
+  $nestedDiv2.appendChild($h3);
+  $nestedDiv2.appendChild($p);
+
+  $nestedDiv1.appendChild($img);
+  $nestedDiv1.appendChild($nestedDiv2);
+
+  $outerDiv.appendChild($nestedDiv1);
+
+  return $outerDiv;
+}
+
+const $row = document.querySelector('.row');
+if (!$row) throw new Error('The $row query failed');
+
+for (const i in pokedex) {
+  // console.log(pokedex[i]);
+  $row.appendChild(renderPokemon(pokedex[i]));
+}
```

Comment:
Technically works because an array is an object, but `for..in` loops are reserved for objects. A better choice would be a `for..of` loop
```suggestion
for (const pokemon of pokedex) {
  $row.appendChild(renderPokemon(pokemon));
}
```